viadot/sources/customer_gauge.py

```diff
@@ -0,0 +1,250 @@
+from viadot.config import local_config
+from viadot.exceptions import CredentialError, APIError
+
+from prefect.utilities import logging
+
+logger = logging.get_logger()
+
+
+class CustomerGauge:
+    API_URL = "https://api.eu.customergauge.com/v7/rest/sync/"
+
+    def __init__(
+        self,
+        endpoint: Literal["responses", "non-responses"] = None,
+        credentials: Dict[str, Any] = None,
+    ):
+        """
+        A class to connect and download data using Customer Gauge API.
+        Below is the documentation for each of this API's gateways:
+            Responses gateway https://support.customergauge.com/support/solutions/articles/5000875861-get-responses
+            Non-Responses gateway https://support.customergauge.com/support/solutions/articles/5000877703-get-non-responses
+
+        Args:
+            endpoint (Literal["responses", "non-responses"]): Indicate which endpoint to connect. Defaults to None.
+            credentials (Dict[str, Any], optional): Credentials to connect with API containing client_id, client_secret. Defaults to None.
+
+        Raises:
+            ValueError: If endpoint is not provided or incorect.
+            CredentialError: If credentials are not provided in local_config or directly as a parameter
+        """
+        self.endpoint = endpoint
+        if endpoint is not None:
+            if endpoint in ["responses", "non-responses"]:
+                self.url = f"{self.API_URL}{endpoint}"
+            else:
+                raise ValueError(
+                    "Incorrect endpoint name. Choose: 'responses' or 'non-responses'"
+                )
+        else:
+            raise ValueError(
+                "Provide endpoint name. Choose: 'responses' or 'non-responses'"
+            )
+
+        if credentials is not None:
+            self.credentials = credentials
+        else:
+            self.credentials = local_config.get("CustomerGauge")
+            if self.credentials is None:
+                raise CredentialError("Credentials not provided.")
+
+    def get_token(self) -> str:
+        """
+        Gets Bearer Token using POST request method.
+
+        Raises:
+            APIError: If token is not returned.
+
+        Returns:
+            str: Bearer Token value
+        """
+        url = "https://auth.EU.customergauge.com/oauth2/token"
+        client_id = self.credentials.get("client_id", None)
+        client_secret = self.credentials.get("client_secret", None)
+
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        body = {
+            "grant_type": "client_credentials",
+            "client_id": client_id,
+            "client_secret": client_secret,
+        }
+        api_response = handle_api_response(
+            url=url, params=body, headers=headers, method="POST"
+        )
+        token = api_response.json().get("access_token")
+
+        if token is None:
+            raise APIError("Token is None. Check your credentials.")
```

Comment:
Change the message e.g( `The token could not be generated. Check your credentials`)

Reply:
The message is still the same.